checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -1506,6 +1519,21 @@ private void checkReassignmentToField(Set<Obligation> obligations, AssignmentNod
           return;
         }
       }
+    } else if (TreeUtils.isConstructor(enclosingMethodTree)) {
+      // Suppress when this assignment is the first write to the private field in this constructor
+      // (later writes/calls are checked at their own CFG points).
+      Element enclosingClassElement =
+          TreeUtils.elementFromDeclaration(enclosingMethodTree).getEnclosingElement();
+      if (ElementUtils.isTypeElement(enclosingClassElement)) {
+        Element receiverElement = TypesUtils.getTypeElement(receiver.getType());
+        if (Objects.equals(enclosingClassElement, receiverElement)) {
+          VariableElement lhsElement = lhs.getElement();
+          if (lhsElement.getModifiers().contains(Modifier.PRIVATE)
```

Comment:
Method `isFirstAssignmentToField` is document to do this test, so move it into the method.